From 7232a2b05ed9c5a5763eab7b735fac7b3ff81c82 Mon Sep 17 00:00:00 2001
From: Sidraya <sidraya.bj@pathpartnertech.com>
Date: Tue, 25 Apr 2023 17:14:57 -0500
Subject: [PATCH 400/508] vxe-vxd: common: Fix structure member initialization
 syntax

modify structure member initialization syntax to fix
ARCH=arm allmodconfig build warnings.

Signed-off-by: Sidraya <sidraya.bj@pathpartnertech.com>
Signed-off-by: Darren Etheridge <detheridge@ti.com>
---
 drivers/media/platform/img/vxe-vxd/decoder/bspp.c |  31 ++---
 drivers/media/platform/img/vxe-vxd/decoder/core.c | 113 ++++++++++++++----
 2 files changed, 105 insertions(+), 39 deletions(-)

diff --git a/drivers/media/platform/img/vxe-vxd/decoder/bspp.c b/drivers/media/platform/img/vxe-vxd/decoder/bspp.c
index 06512bc85c71..82b5f0d93bd8 100644
--- a/drivers/media/platform/img/vxe-vxd/decoder/bspp.c
+++ b/drivers/media/platform/img/vxe-vxd/decoder/bspp.c
@@ -183,35 +183,38 @@ struct bspp_parser_functions {
 
 static struct bspp_parser_functions parser_fxns[VDEC_STD_MAX] = {
 	/* VDEC_STD_UNDEFINED */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_MPEG2 */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_MPEG4 */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_H263 */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_H264 */
-	{ bspp_h264_set_parser_config, bspp_h264_determine_unittype },
+	{ .set_parser_config = bspp_h264_set_parser_config,
+		.determine_unit_type = bspp_h264_determine_unittype },
 	/* VDEC_STD_VC1 */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_AVS */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_REAL */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_JPEG */
 #ifdef HAS_JPEG
-	{ bspp_jpeg_setparser_config, bspp_jpeg_determine_unit_type },
+	{ .set_parser_config = bspp_jpeg_setparser_config,
+		.determine_unit_type = bspp_jpeg_determine_unit_type },
 #else
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 #endif
 	/* VDEC_STD_VP6 */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_VP8 */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_SORENSON */
-	{ NULL, NULL },
+	{ .set_parser_config = NULL, .determine_unit_type = NULL },
 	/* VDEC_STD_HEVC */
-	{ bspp_hevc_set_parser_config, bspp_hevc_determine_unittype },
+	{ .set_parser_config = bspp_hevc_set_parser_config,
+		.determine_unit_type = bspp_hevc_determine_unittype },
 };
 
 /*
diff --git a/drivers/media/platform/img/vxe-vxd/decoder/core.c b/drivers/media/platform/img/vxe-vxd/decoder/core.c
index 237e7f937751..9d1b560610df 100644
--- a/drivers/media/platform/img/vxe-vxd/decoder/core.c
+++ b/drivers/media/platform/img/vxe-vxd/decoder/core.c
@@ -283,34 +283,97 @@ core_common_bufs_getsize(struct core_stream_context *core_str_ctx,
 
 static struct core_std_spec_operations std_specific_ops[VDEC_STD_MAX - 1] = {
 	/* AllocPicture  FreePicture  AllocSeq   FreeSeq   BufsGetSize  IsStreamResourceSuitable */
-	/* MPEG2   */ { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-	/* MPEG4   */ { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-	/* H263    */ { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
+	/* MPEG2   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* MPEG4   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* H263   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* H264   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = core_common_bufs_getsize,
+		.is_stream_resource_suitable = NULL},
+
+	/* VC1   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* AVS   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* REAL   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* JPEG   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* VP6   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* VP8   */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
+
+	/* SORENSON */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
 
-	/* H264    */ { NULL,       NULL,      NULL,   NULL,  core_common_bufs_getsize,      NULL},
-
-	/* VC1    */  { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-
-	/* AVS    */  { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-
-	/* REAL    */ { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-
-	/* JPEG    */ { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-
-	/* VP6    */  { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-
-	/* VP8    */  { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
-
-	/* SORENSON    */ { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
 #ifdef HAS_HEVC
-	/* HEVC     */ { core_alloc_hevc_specific_pict_buffers,
-			 core_free_hevc_picture_resource,
-			 core_alloc_hevc_specific_seq_buffers,
-			 core_free_hevc_sequence_resource,
-			 core_hevc_bufs_get_size,
-			 core_is_hevc_stream_resource_suitable},
+	/* HEVC*/ { .alloc_picture_buffers = core_alloc_hevc_specific_pict_buffers,
+		.free_picture_resource = core_free_hevc_picture_resource,
+		.alloc_sequence_buffers = core_alloc_hevc_specific_seq_buffers,
+		.free_sequence_resource = core_free_hevc_sequence_resource,
+		.bufs_get_size = core_hevc_bufs_get_size,
+		.is_stream_resource_suitable = core_is_hevc_stream_resource_suitable},
 #else
-	/* HEVC    */ { NULL,       NULL,      NULL,   NULL,  NULL,      NULL},
+	/* HEVC */ { .alloc_picture_buffers = NULL,
+		.free_picture_resource = NULL,
+		.alloc_sequence_buffers = NULL,
+		.free_sequence_resource = NULL,
+		.bufs_get_size = NULL,
+		.is_stream_resource_suitable = NULL},
 #endif
 };
 
-- 
2.41.0

